devicetypes/sky-nie/evalogik-door-window-sensor.src/evalogik-door-window-sensor.groovy

```diff
@@ -0,0 +1,615 @@
+
+private static getCommandClassVersions() {
+	[
+		0x30: 2,  // SensorBinary
+		0x31: 5,  // SensorMultilevel
+		0x55: 1,  // TransportServices
+		0x59: 1,  // AssociationGrpInfo
+		0x5A: 1,  // DeviceResetLocally
+		0x5E: 2,  // ZwaveplusInfo
+		0x6C: 1,  // Supervision
+		0x70: 1,  // Configuration
+		0x71: 3,  // Notification
+		0x72: 2,  // ManufacturerSpecific
+		0x73: 1,  // Powerlevel
+		0x7A: 2,  // FirmwareUpdateMd
+		0x80: 1,  // Battery
+		0x84: 1,  // WakeUp
+		0x85: 2,  // Association
+		0x86: 1,  // Version
+		0x8E: 2,  // MultChannelAssociation
+		0x87: 1,  // Indicator
+		0x9F: 1   // Security 2
+	]
+}
+
+private canReportBattery() {
+	return state.refreshSensors || (!isDuplicateCommand(state.lastBatteryReport, (12 * 60 * 60 * 1000)))
+}
+
+private getPendingChanges() {
+	return configParams.count { "${it.value}" != "${getParamStoredValue(it.num)}" }
+}
+
+private getParamStoredValue(paramNum) {
+	return safeToInt(state["configParam${paramNum}"] , null)
+}
+
+// Sensor Types
+private static getTempSensorType() { return 1 }
+private static getLightSensorType() { return 5 }
+
+// Configuration Parameters
+private getConfigParams() {
+	[
+		batteryReportThresholdParam,
+		lowBatteryAlarmReportParam,
+		sensorModeWhenClosedParam,
+		delayReportSecondsWhenClosedParam,
+		delayReportSecondsWhenOpenedParam,
+		minTemperatureOffsetParam,
+		minHumidityOffsetParam,
+		temperatureUpperWatermarkParam,
+		temperatureUpperControlParam,
+		temperatureLowerWatermarkParam,
+		temperatureLowerControlParam,
+		humidityUpperWatermarkParam,
+		humidityUpperControlParam,
+		humidityLowerWatermarkParam,
+		humidityLowerControlParam,
+		switchTemperatureUnitParam,
+		temperatureOffsetParam,
+		humidityOffsetParam,
+		associationGroupSettingParam
+	]
+}
+
+private getBatteryReportThresholdParam() {
+	return getParam(1, "Battery report threshold(1% - 20%)", 1, 10, null,"1..20")
+}
+
+private getLowBatteryAlarmReportParam() {
+	return getParam(2, "Low battery alarm report(5% - 20%)", 1, 5, null, "5..20")
+}
+
+private getSensorModeWhenClosedParam() {
+	return getParam(3, "State of the sensor when the magnet closes the reed", 1, 0, sensorModeWhenCloseOptions)
+}
+
+private getDelayReportSecondsWhenClosedParam() {
+	return getParam(4, "Delay in seconds with ON command report(door closed)", 2, 0, null, "0..3600")
+}
+
+private getDelayReportSecondsWhenOpenedParam() {
+	return getParam(5, "Delay in seconds with OFF command report(door open)", 2, 0, null, "0..3600")
+}
+
+private getMinTemperatureOffsetParam() {
+	return getParam(6, "Minimum Temperature change to report(0.5℃/0.9°F - 5.0℃/9°F)", 1, 10, null, "5..50")
+}
+
+private getMinHumidityOffsetParam() {
+	return getParam(7, "Minimum Humidity change to report(5% - 20%)", 1, 10, null, "5..20")
+}
+
+private getTemperatureUpperWatermarkParam() {
+	return getParam(8, "Temperature Upper Watermark value(0,Disabled; 1℃/33.8°F-50℃/122.0°F)", 2, 0, null, "0..50")
+}
+
+private getTemperatureUpperControlParam() {
```

Comment:
Could all the parameters related to associations group may be omitted ? Instead of it device might be pre-configured.

Reply:
Ok, I delete all the parameters related to associations group except  "associationGroupSettingParam".